tcf_website/static/schedule/parse_time.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable */
+
+function stringTimeToInt(stringTime) {
+    let ret = 0;
+    parsed_time = stringTime.split(":");
+    ret += 60 * parseInt(parsed_time[0]);
+    if(parsed_time[1][2] == 'p' && parsed_time[0] != "12") {
+        ret += 720;
+    }
+    minute_string = parsed_time[1].substr(0, 2);
+    ret += parseInt(minute_string);
+
+    return ret;
+}
+
+function parseTime(time_string) {
+    time_string = time_string.split(","); //split lab times / discussion times / lecture times
+    let weekday_meeting_times = Array.apply(null, Array(5)).map(function () {}) //index 0 = monday, index 1 = tuesday, index 2 = wednesday, index 3 = thursday, index 4 = friday
+    for(let i=0; i<weekday_meeting_times.length;i++)
+    {
+        weekday_meeting_times[i] = [];
+    }
+
+
+    for (let j = 0; j < time_string.length; j++)
+    {
+        let meetingTime = time_string[j].split(" "); //creates an array that contains[days_of_week, class_start_time, -, class_end_time]
+        let days_of_week = meetingTime[0];
+
+
+        while(days_of_week!="")
+        {
+            //adds both the start and end time for a particular class to a weekday
+            let class_time_pair = [stringTimeToInt(meetingTime[1]), stringTimeToInt(meetingTime[3])];
+            
+            if(days_of_week.includes("Mo"))
+            {
+                weekday_meeting_times[0].push(class_time_pair);
+                days_of_week = days_of_week.replace("Mo","");
+            }
+            if(days_of_week.includes("Tu"))
+            {
+                weekday_meeting_times[1].push(class_time_pair);
+                days_of_week = days_of_week.replace("Tu","");
+            }
+            if(days_of_week.includes("We"))
+            {
+                weekday_meeting_times[2].push(class_time_pair);
+                days_of_week = days_of_week.replace("We","");
+            }
+            if(days_of_week.includes("Th"))
+            {
+                weekday_meeting_times[3].push(class_time_pair);
+                days_of_week = days_of_week.replace("Th","");
+            }
+            if(days_of_week.includes("Fr"))
+            {
+                weekday_meeting_times[4].push(class_time_pair);
+                days_of_week = days_of_week.replace("Fr","");
+            }
+        }
+    }
+    return weekday_meeting_times;
+}
+
+function consolidateTimes(times) {
+    selected_classes_meeting_times = [[],[],[],[],[]]
+    for (var selected_class = 0; selected_class < times.length; selected_class++) {
```

Comment:
Also, once again, the organization of this data makes think function very odd. Wouldn't it make more sense, for example, to have a list of selected class identifiers that map to meeting times explicitly, rather than implicityl through an array structure?